[Feature] 오늘의 승부예측 경기 정보 API 연동 완료

sihyun92 opened the pull request:
### 📌 개요

<!-- PR과 관련있는 Issue를 closed 해주세요 -->

closed #42

작업한 내용에 대해 간단히 작성해주세요.

### ✅ 작업내용

- Today라는 컴포넌트를 새로 생성해서 CSR 영역과 SSR 영역 분리
- TodayPage 컴포넌트에 Today 컴포넌트 렌더링 처리
- ScoreList로 받아온 경기 정보 배열 데이터 map으로 배열 원소 렌더링 처리
- 당일 경기 정보에 맞춰서 팀로고 렌더링 처리
- React-Query로 API 호출 완료 

### 📝 공유사항

팀원들에게 공유할 사항이나 변경 사항들을 작성해주세요.

- 작동은 됩니다만 팀 로고를 출력하는 함수 안에 분기 로직이 너무 많아 리팩토링이 필요합니다! 좋은 의견 있으시면 한 수 가르쳐주세요~! ㅎㅎㅎㅎ

sihyun92 committed:
Merge branch 'dev' of github.com:baseball-project/client into feature/today

sihyun92 committed:
Fix: 인천SSG로 변경

sihyun92 committed:
Merge branch 'dev' of github.com:baseball-project/client into feature/today

doitidey committed:
Feature: 버튼 클릭시 input 연결

doitidey committed:
Feature: 이미지 미리보기

doitidey committed:
Feature: 선택된 카드 선택하기 호버 기능 제외

doitidey committed:
Feature: 카드 사용 컴포넌트에서 프랍 추가

doitidey committed:
Refactor: 단계 store 이름 변경

doitidey committed:
Feature: 닉넴 중복체크 api 추가

doitidey committed:
Style: 스토어 명 변경

doitidey committed:
Feature: 폼 데이터 zustand로 관리 후 넘기기 위한 작업

sihyun92 committed:
Feat: user-select 속성 추가
- .live 내 텍스트 블록처리 방지

sihyun92 committed:
Feat: 오늘의 승부예측 경기 조회 api 구현

sihyun92 committed:
Feat: 오늘의 승부예측 경기 목록 함수 구현 시작

doitidey committed:
Feature: 입력값 form data로 받아온 후 store 저장

sihyun92 committed:
Refactor: Today 컴포넌트 리팩토링
- 기존의 TodayPage에 ScoreList, DateSection을 직접 렌더링하는 방식에서 Today 컴포넌트에 컴포넌트를 삽입해서 Today 컴포넌트를 TodayPage 컴포넌트에 렌더링하는 방식으로 변경
- 페이지 컴포넌트 자체는 SSR 방식이기 때문에 use client를 사용하는 것은 바람직하지 않아서 방식 변경

sihyun92 committed:
Design: Today 컴포넌트에 상하 마진 값 적용
- 기존 TodayPage.scss의 속성을 Today.scss로 리팩토링

sihyun92 committed:
Feat: 당일 경기 정보 API 연동 1차 완료
- API 상의 경기 날짜 렌더링 완료
- 각 경기 정보 렌더링 완료
- 로고 이미지는 경기 정보에 따라 팀에 맞춰서 렌더링하게 끔 함수 작성 -> 추후 리팩토링 필요

sihyun92 committed:
Docs: getHomeTeamSrc 코멘트 추가

sihyun92 committed:
Feat: 당일 경기 정보 API 배열 map으로 배열 요소 렌더링
- Today 컴포넌트에서 받아온 경기 정보 배열 요소를 map 메서드로 렌더링

sihyun92 committed:
Refactor: Today 컴포넌트로 리팩토링 후 렌더링

doitidey commented on src/components/today/ScoreListItem.tsx:
```diff
@@ -38,10 +67,61 @@ function ScoreListItem({ home, away, homeImg, awayImg }: ScoreListItemProps) {
   }, [selectAway]);
 
+  // 리팩토링 예정 => 조건문이 너무 많음, 같은 로직의 함수가 중복 됨
+  const getHomeTeamSrc = (data: string) => {
+    if (homeTeam.teamName === "LG 트윈스") {
+      return TeamImgHref.twins;
+    } else if (homeTeam.teamName === "삼성 라이온즈") {
+      return TeamImgHref.lions;
+    } else if (homeTeam.teamName === "한화 이글스") {
+      return TeamImgHref.eagles;
+    } else if (homeTeam.teamName === "KIA 타이거즈") {
+      return TeamImgHref.tigers;
+    } else if (homeTeam.teamName === "키움 히어로즈") {
+      return TeamImgHref.heroes;
+    } else if (homeTeam.teamName === "롯데 자이언츠") {
+      return TeamImgHref.gients;
+    } else if (homeTeam.teamName === "SSG 랜더스") {
+      return TeamImgHref.landers;
+    } else if (homeTeam.teamName === "두산 베어스") {
+      return TeamImgHref.bears;
+    } else if (homeTeam.teamName === "KT 위즈") {
+      return TeamImgHref.wiz;
+    } else if (homeTeam.teamName === "NC 다이노스") {
+      return TeamImgHref.dinos;
+    }
+    return data;
+  };
```
비교를 위한 조건문이 다중으로 사용될 경우 `switch`를 활용하시는 것도 가독성을 위해서 좋을 것 같습니다!
그리고 `TeamImgHref`, `getHomeTeamSrc`, `getAwayTeamSrc` 와 같은 함수는 다른 파일로 관리해주는 것이 더 좋아보입니다!

doitidey reviewed (commented):
연휴 도중에도 수고 많으셨습니다! 코드 훑어보고 코멘트 남겼으니 참고 부탁드려요~! 👍

sihyun92 committed:
Feat: QueryClientProvider 컴포넌트 추가
- react-query 인스턴스 생성
- QueryClientProvider 컴포넌트 추가
- layout 컴포넌트 client 컴포넌트로 전환

sihyun92 committed:
Refactor: 팀로고 경로추가 함수 리팩토링
- 팀로고 함수 ScoreListItem -> getLogo로 분리
- switch 문 조건에 data로 변경

sihyun92 committed:
Refactor: 경기 시간 함수 분리
- Today 컴포넌트에 있는 함수를 getGameTime 파일로 분리
- ScoreListItem에서 호출 완료
- DateSection 컴포넌트에도 호출 예정

sihyun92 committed:
Feat: React-Query로 오늘의 승부예측 조회 호출 완료
- 기존 useCallback과 useEffect로 호출하는 함수 제거
- useQuery로 호출 완료
- 호출 이후 렌더링하는 과정에서 성능 이슈 발생 (해결 중)

sihyun92 committed:
Refactor: 컴포넌트 내부의 함수 리팩토링
- getHomeLogo, getAwayLogo 함수 -> getLogo 파일에 리팩토링
- gameDate 함수 -> getGameTime 파일에 리팩토링

doitidey committed:
Feature: 팀카드 타입 구조 변경 및  단일 카드 클릭 여부 판단 후 css 변경 로직 추가

doitidey committed:
Feature: 클릭된 카드 zustand통해 step2로 전송

sihyun92 committed:
Feat: 페이지 스크롤바 삭제
- 전체 페이지 스크롤바 삭제

sihyun92 committed:
Merge branch 'dev' of github.com:baseball-project/client into feature/today

sihyun92 committed:
Merge branch 'feature/signup' of github.com:baseball-project/client into feature/today

sihyun92 committed:
Fix: useQuery data 변수 삭제

sihyun92 committed:
Feat: 오늘의 승부예측 댓글 api 함수 작성
